File: src/main/java/me/zeroeightsix/kami/module/modules/bewwawho/misc/ConsoleSpam.java
package me.zeroeightsix.kami.module.modules.bewwawho.misc;

import me.zeroeightsix.kami.module.Module;
import me.zeroeightsix.kami.event.events.PacketEvent;
import me.zeroeightsix.kami.command.Command;
import me.zeroeightsix.kami.util.zeroeightysix.Wrapper;

import me.zero.alpine.listener.EventHandler;
import me.zero.alpine.listener.Listener;

import net.minecraft.network.play.client.CPacketPlayerTryUseItemOnBlock;
import net.minecraft.network.play.client.CPacketUpdateSign;
import net.minecraft.util.math.BlockPos;
import net.minecraft.tileentity.TileEntitySign;

/**
 * Created on 16 December by 0x2E | PretendingToCode
 */

@Module.Info(name = "ConsoleSpam", description = "Spams Spigot consoles by sending invalid UpdateSign packets", category = Module.Category.MISC)
public class ConsoleSpam extends Module {

    @Override
    public void onEnable() {
        Command.sendChatMessage("[ConsoleSpam] Every time you right click a sign, a warning will appear in console.");
        Command.sendChatMessage("[ConsoleSpam] Use an autoclicker to automate this process.");
    }

    @EventHandler
    public Listener<PacketEvent.Send> sendListener = new Listener<>(event -> {
        if (event.getPacket() instanceof CPacketPlayerTryUseItemOnBlock) {
            BlockPos location = ((CPacketPlayerTryUseItemOnBlock) event.getPacket()).getPos();

            Wrapper.getPlayer().connection.sendPacket(new CPacketUpdateSign(location, new TileEntitySign().signText));
        }
    });
}

File: src/main/kotlin/com/lambda/client/module/modules/misc/ConsoleSpam.kt
package com.lambda.client.module.modules.misc

import com.lambda.client.event.events.PacketEvent
import com.lambda.client.module.Category
import com.lambda.client.module.Module
import com.lambda.client.util.text.MessageSendHelper.sendChatMessage
import com.lambda.client.util.threads.safeListener
import net.minecraft.network.play.client.CPacketPlayerTryUseItemOnBlock
import net.minecraft.network.play.client.CPacketUpdateSign
import net.minecraft.tileentity.TileEntitySign

object ConsoleSpam : Module(
    name = "ConsoleSpam",
    description = "Spams Spigot consoles by sending invalid UpdateSign packets",
    category = Category.MISC
) {
    init {
        onEnable {
            sendChatMessage("$chatName Every time you right click a sign, a warning will appear in console.")
            sendChatMessage("$chatName Use an auto clicker to automate this process.")
        }

        safeListener<PacketEvent.Send> {
            if (it.packet !is CPacketPlayerTryUseItemOnBlock) return@safeListener
            connection.sendPacket(CPacketUpdateSign(it.packet.pos, TileEntitySign().signText))
        }
    }
}